gpflow/conditionals.py

```diff
@@ -157,12 +159,17 @@ def uncertain_conditional(Xnew_mu, Xnew_var, feat, kern, q_mu, q_sqrt, *,
         # This is not implemented as this feature is only used for plotting purposes.
         raise NotImplementedError
 
+    pXnew = Gaussian(Xnew_mu, Xnew_var)
+
     num_data = tf.shape(Xnew_mu)[0]  # number of new inputs (N)
+    num_ind = tf.shape(q_mu)[0]  # number of inducing points (M)
     num_func = tf.shape(q_mu)[1]  # output dimension (D)
 
+    # mean_function = mean_function or mean_functions.Zero(output_dim=num_func)
+
     q_sqrt_r = tf.matrix_band_part(tf.transpose(q_sqrt, (2, 0, 1)), -1, 0)  # D x M x M
 
-    eKuf = tf.transpose(feat.eKfu(kern, Xnew_mu, Xnew_var))  # M x N
+    eKuf = tf.transpose(expectation(pXnew, (kern, feat))) # M x N (psi1)
```

Comment:
I believe the feat.eKfu and eKufKfu methods were added solely to support this; now that there is a general `expectation()`, could you remove them from `features.py` again?